apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c

```diff
@@ -0,0 +1,1128 @@
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile,
+	sizeof(gatt_profile) / sizeof(ble_server_gatt_t)
+};
+
+static void set_scan_filter(ble_scan_filter *filter, uint8_t *raw_data, uint8_t len, bool whitelist_enable, uint32_t scan_duration)
+{
+	memset(filter, 0, sizeof(ble_scan_filter));
+	if (raw_data != NULL && len > 0) {
+		memcpy(filter->raw_data, raw_data, len);
+		filter->raw_data_length = len;
+	}
+
+	filter->scan_duration = scan_duration;
+	filter->whitelist_enable = whitelist_enable;
+}
+
+static void utc_blemanager_init_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);//not server role
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(&server_config), BLE_MANAGER_SUCCESS);//server role
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_init_n(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(&server_config), BLE_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(&server_config), BLE_MANAGER_ALREADY_WORKING);
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_deinit_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);//not server role
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(&server_config), BLE_MANAGER_SUCCESS);//server role
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_deinit_n(void)
+{
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_INVALID_STATE);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_mac_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);
+	uint8_t mac[BLE_BD_ADDR_MAX_LEN];
+	TC_ASSERT_EQ_CLEANUP("ble_manager_get_mac_addr", ble_manager_get_mac_addr(mac), BLE_MANAGER_SUCCESS, );
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_mac_n(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);
+	TC_ASSERT_EQ("ble_manager_get_mac_addr", ble_manager_get_mac_addr(NULL), BLE_MANAGER_INVALID_ARGS);
+	TC_ASSERT_EQ("ble_manager deinit", ble_manager_deinit(), BLE_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_blemanager_get_bonded_device_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);
+	ble_bonded_device_list dev_list[BLE_MAX_BONDED_DEVICE] = { 0, };
+	uint16_t dev_count = 0;
+	ble_client_ctx *ctx;
+	ctx = ble_client_create_ctx(&client_config);
+	uint8_t dummy[] = {0x00, 0x1E, 0xC0, 0x04, 0x9F, 0xF3};
```

Comment:
Could you move all initializations to the start of this API

Reply:
The intuition is that, if the init fails the whole testcase fails, so the variables initialized above init will be useless in such case. So they come after init